ocaml/xapi/xapi_vm.ml

```diff
@@ -238,6 +238,32 @@ let assert_memory_constraints ~__context ~vm platformdata =
    same VM twice during memory calculations to determine whether a given VM can start on a particular host..
 *)
 
+let update_platform_secureboot ~__context ~self platform =
+  if List.assoc "secureboot" platform = "auto" then
+    let t = ("secureboot", string_of_bool (Db.Pool.get_uefi_certificates ~__context ~self <> "")) in
+    List.append [t] (List.remove_assoc "secureboot" platform)
+  else
+    platform
+
+let save_uefi_certificates_to_dir ~__context ~self =
+  let err () =
+    debug "Error writing uefi certificates to file";
```

Comment:
This would benefit from including the specifics: what file is missing in what directory.